en/docs/references/extend/federation/write-a-custom-oauth2-federated-authenticator.md

```diff
@@ -0,0 +1,84 @@
+|[protected void processAuthenticationResponse(HttpServletRequest request,HttpServletResponse response, AuthenticationContext context)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L116-L145)|Implements the logic of the custom federated authenticator.|
+|[protected void buildClaims(AuthenticationContext context, String userInfoString)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L147-L183)|Implements the logic for extracting the subject from claims and for other claim mappings.|
+
+---
+
+## Deploy the custom OAuth2 federated authenticator in WSO2 IS
+To deploy the custom OAuth2 federated authenticator:
+1. Open a terminal, navigate to the root of your project, and compile the service by running the following command:
+    ``` xml
+    mvn clean install
+    ```
+2. Copy the compiled `jar` file inside `/target` folder and paste it to the `<IS_HOME>/repository/components/dropins`.
+
+---
+## Configure the Identity Server
+
+This section guides you on how to configure the identity server to use the custom OAuth2 federated authenticator.
+### Configure custom OAuth2 federated authenticator
+
+To configure the OAuth2 custom authenticator: 
+1. Initially, [register a new identity provider]({{base_path}}/guides/identity-federation/add-idp/).
+2. Expand **Federated Authenticators > Custom OAuth2 Authenticator Configuration** and configure it as follows:
+
+    - Select the **Enable** and **Default** check boxes.
+    - **Client Id** - The client ID value generated by the external OAuth2 IdP.
+    - **Client Secret** - The client secret value generated by the external OAuth2 IdP.
+    - **Callback Url** - https://localhost:9443/commonauth
+    - **Authorization Endpoint URL** - https://kauth.kakao.com/oauth/authorize
+    - **Token Endpoint URL** - https://kauth.kakao.com/oauth/token
+    - **User Information Endpoint URL** - https://kapi.kakao.com/v2/user/me
+
+
+    !!!info
+         Here, the Client Id and Client Secret are the values of Kakao.
```

Comment:
Don't we need to point here the guidelines to take these values from Kakao?

Reply:
At a point where writing a custom OAuth2 authenticator, we expect the developer to know the basics of the external OAuth2 IdP. Therefore, we can rephrase this to be more generic with,
`Here, the Client Id and Client Secret are the values generated by the external OAuth2 IdP.`
What do you think?

Reply:
Changed with https://github.com/wso2/docs-is/pull/3421